File: python/problem_003.py
#!/usr/bin/env python3
import time

def blurb():
  print("""

  The prime factors of 13195 are 5, 7, 13 and 29.

  What is the largest prime factor of the number 600851475143 ?

  """)

def main():
  UPPER_LIMIT = 600851475143

  start = time.time()

  FACTOR = 2
  while FACTOR * FACTOR < UPPER_LIMIT:
    while UPPER_LIMIT % FACTOR == 0:
      UPPER_LIMIT = UPPER_LIMIT / FACTOR
    FACTOR += 1

  RESULT = UPPER_LIMIT

  end   = time.time() - start
  print("Result: %s"      % RESULT)
  print("Completed in %s" % end)

if __name__ == '__main__':
  main()

File: javascript/problem_003.js
#!/usr/bin/env node

function printBlurb() {
    var blurb = `
    The prime factors of 13195 are 5, 7, 13 and 29.

    What is the largest prime factor of the number 600851475143 ?
    `
    console.log(blurb)
}

function problem_003(primeFactorOf) {
    var factorToTest = 2
    while (factorToTest * factorToTest < primeFactorOf) {
        while (primeFactorOf % factorToTest == 0) {
            primeFactorOf = primeFactorOf / factorToTest
        }
        factorToTest += 1
    }
    return primeFactorOf
}

printBlurb()
console.time('Completed in')
result = problem_003(600851475143)
console.log(`Result: ${result}`)
console.timeEnd('Completed in')
